### DISCUSSION TITLE
Potential issue added for 4 channel Hoymiles inverter in Wiki

### DISCUSSION THREAD

**Discussion by MalteSchm** [hoylabs/OpenDTU-OnBattery#678] [General]
I thought I'd make a note here about the potential issue described here:
https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/HOYMILES

I hope that this gets more reach this way.

I measured my setup today and found about 1.1A current that flows over this ground connection at full load.

**Comment by Manos1966** [comment]
Hi @MalteSchm 
guck Mal bitte ob es sinnvoll ist auch hier etwas zu addieren:
https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/Kabell%C3%A4nge%28n%29-zwischen-Batterie-und-Wechselrichter

**Reply by MalteSchm** [reply]
Ich habe die Seite mal überarbeitet und die Gründe für gleiche Kabellängen aufgeschrieben. Deinen Text habe ich gelassen aber ich denke man sollte schauen was davon noch nötig ist.

**Reply by Manos1966** [reply]
> Deinen Text habe ich gelassen aber ich denke man sollte schauen was davon noch nötig ist.

muss du nicht. Passe alles nach besten Wissen und Gewissen so wie du es für sinnvoll halst 😉

**Reply by MalteSchm** [reply]
Gut erledigt

**Comment by spcqike** [comment]
Wie hast du denn ermittelt wie viel Strom da wo fließt? 

Und, so recht verstehe ich es noch nicht :o 

> I […] found about 1.1A current that flows over this ground connection at full load

„This“? diese eine? Du hast doch 4 GNDs zur Batterie.

**Reply by MalteSchm** [reply]
Ich habe mit einem Zangenamperemeter gemessen was auf Kanal 1/2 auf + rein und auf - raus ging. Das geht indem man über alle Zuleitungen misst. Diese Summe hätte 0 müssen. War sie aber nicht. Das gleiche habe ich auf Kanal 3/4 gemacht. 
Auf Kanal 1/2 gingen quasi 1.1A mehr rein als raus kamen. Auf Kanal 3/4 war es umgekehrt. (Aktuelle Messwerte waren glaube ich 1.4A und -0.8A)

**Reply by spcqike** [reply]
Das macht Sinn. 

Hast du ne Idee woran das bei dir im konkreten liegt und wie du es beheben willst? 

wenn auf 3/4 mehr raus geht, scheint ja dein leitungswiderstand dort etwas geringer zu sein. Also die GND Leitungen von 1/2 etwas kürzen? Bzw erstmal die klemmstellen / Verbinder sichten und ggf nachziehen? 

Nachtrag 
Wie hast du es denn an die Batterie geklemmt? Mit ner busbar? Wenn ja kann man da vielleicht auch dort die Positionierung / tauschen etwas an dem sich ergebenden Widerstand machen

**Reply by SW-Niko** [reply]
Hallo @MalteSchm ,
hast du das in der nähe vom maximalen Strom gemessen?
Ich denke das bei kleinen Strömen ein Unterschied nicht viel ausmacht. Erst wenn der Hoymiles in die Strom-Begrenzung geht verlierst du etwas an Leistung. I vermute das hier die Unterschiede geringer sind, weil die Leitungen einen positiven Temperaturkoeffizienten haben.
Also hochohmiger werden und damit etwas für zum Ausgleich der Ströme beitragen.
Wenn du 1A unterschied hast dann sind das 0.5A zu viel in einem Teil. Das sind ca 4.5% von 11A oder 4.5% vom Widerstand.
Viel Verbesserungspotential sehe ich nicht.

**Reply by MalteSchm** [reply]
Gemessen wurde bei maximaler Leistung (1500W). 

> Hast du ne Idee woran das bei dir im konkreten liegt und wie du es beheben willst?

Man kann versuchen die Ströme auszugleichen. Aber das ganze ist ein ziemlich fundamentales Problem. Deshalb auch mein Post hier.
Möglich wäre die Stecker von den Minusleitungen abzuschneiden und diese mit Crimpkontaken auf einen Massepunkt zu führen. Aber schön ist das nicht.
Man muss halt auch festhalten das der WR erst mal so auch arbeitet. Hat er jetzt ein Jahr getan. 

> Nachtrag Wie hast du es denn an die Batterie geklemmt? Mit ner busbar? Wenn ja kann man da vielleicht auch dort die Positionierung / tauschen etwas an dem sich ergebenden Widerstand machen

Alle Kabel gehen über die MC4 Stecker an einen Massepunkt. Aber auch die Stecker werden unterschiedliche Widerstände haben.

**Comment by Manos1966** [comment]
@MalteSchm @spcqike @SW-Niko 
Der Unterschied zwischen Batterie und Solar Paneele ist, das die Paneele Dioden haben, richtig?
Wenn man also die vier DC minus mit Dioden sichern wuerde, waere das "Problem" beseitig, oder?

Wobei es gar kein "Problem" ist, wenn ich richtig verstanden habe ("Man muss halt auch festhalten das der WR erst mal so auch arbeitet. Hat er jetzt ein Jahr getan.")

**Reply by MalteSchm** [reply]
Der Unterschied ist dass Paneele getrennte Stromkreise darstellen und die Batterie nicht. 
Dioden helfen nicht

Wie gravierend das ist, ist mir nicht klar. Ich habe meine Anlage deswegen nicht gleich ausgeschaltet. Ob ich nochmal mit HM 1500 bauen würde weiß nicht nicht. Evtl eher ein  800 nehmen

**Reply by SW-Niko** [reply]
Ja genau, Dioden in den DC minus Leitungen helfen hier nicht. Auf der einen Seite verbindet der Hoymiles die Massen und auf der anderen Seite die Batterie. Also sind die Dioden wie die "Messwiderstände im Hoymiles" dann parallel geschaltet und es kann zu einer ungleichmäßigen Stromverteilung kommen. 
Ich sehe kein Problem im Normalbetrieb solange die Unterschiede keiner als 10% sind. 
Kritisch sehe ich eher die Sonderfälle. Z.B. Bei 2 Eingängen fällt eine Masse Leitung aus. Dann fließt theoretisch der doppelte Strom über die Masseverbindung des 2ten Kanals. Greift dann notfalls die Strombegrenzung ein? Und was ist dann mit 4 Kanälen?
Um eine fundierte Aussage zu machen müsse ich den internen Aufbau der Hoymiles besser kennen. Sorry.

**Reply by SW-Niko** [reply]
Noch was....
Ich würde für die Masseleitungen zwischen Batterie und Inverter keine einzelnen Sicherungen oder Trennschalter verbauen.
Also wenn  ....  dann erst alle Leitungen zusammenführen und dann eine Sicherung und/oder Trennschalter.

**Reply by spcqike** [reply]
> Dann fließt theoretisch der doppelte Strom über die Masseverbindung des 2ten Kanals. Greift dann notfalls die Strombegrenzung ein? Und was ist dann mit 4 Kanälen?

wie hier https://github.com/helgeerbe/OpenDTU-OnBattery/issues/697#issuecomment-1963572832 geschildert, greift da wohl nichts. 

Bzw, wenn ein MPPT mehrere Eingänge hat, kann ein einzelner Eingang scheinbar 1/n_anzahl_MPPT der Gesamtleistung des wechselrichters ziehen. Gemessen wird wohl pro Eingang, gedrosselt pro MPPT. 

Bei meinem HMS-2000 (4 Eingänge, 4 MPPT) hab ich es am Labor Netzteil getestet, da liefen nur die 500W über einen Eingang. 

hhoebel hingegen berichtet von seinem HMT-2250 (6 Eingänge, 3 MPPT), das ein einzelner Eingang bis zu 750W verarbeiten kann. Also genau 1/3 der maximalen Leistung.

**Comment by Manos1966** [comment]
> Noch was.... Ich würde für die Masseleitungen zwischen Batterie und Inverter keine einzelnen Sicherungen oder Trennschalter verbauen. Also wenn .... dann erst alle Leitungen zusammenführen und dann eine Sicherung und/oder Trennschalter.

addiere ich in die Wiki
